### PR TITLE
Update continue onboarding to use embedded component

### LINKED ISSUE #9357: Handle continue onboarding links in the embedded component
### Description
In https://github.com/Automattic/woocommerce-payments/pull/9251, we implemented embedded component in MOX. We need to use it in account links when the merchant wants to continue onboarding from the connect page.

### PR DESCRIPTION
**Pull request by oaratovskyi**
Fixes #9357 

#### Changes proposed in this Pull Request

* Redirect the merchants who haven't completed onboarding to the page that is similar MOX, Stripe Embedded step
* I took @ismaeldcom PR https://github.com/Automattic/woocommerce-payments/pull/6934 for inspiration

#### Prerequisites

* Run `npm run watch`
* Update the version string in `woocommerce-payments.php` to `8.2.0`
* Clear any existing account from your store

0. Start new onboarding via the Connect page
0. Leave the MOX at step 2, planned income and time to go live (click X in the top right corner).
0. Continue onboarding from the Connect page again.
0. You should arrive at start of the MOX - your previous choices won't be remembered
0. Now proceed to the embedded step. Start the Stripe onboarding, maybe click through a few screens but don't complete it. Then leave the MOX. (either close the page or click `X` in the top right corner)
0. From the connect page, continue setting up your account - you should arrive to the new page `payments/onboarding/kyc`. You can try proceeding step by step, leaving on each step and then checking you are redirected to the right step of embedded onboarding.
0. Proceed with setting up progressive account
0. Now try to set up deposits - you should be redirected to the Stripe KYC (since it's separate issue).

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9380` or branch name `update/9357-continue-onboarding-links-in-embedded-component` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/9357-continue-onboarding-links-in-embedded-component) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 05f322e327113157a2234908fd3988fb2d22ac26
- Build time: 2024-09-06 12:24:34 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/onboarding/kyc/index.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import React, { useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Page from 'wcpay/components/page';
+import { OnboardingContextProvider } from 'wcpay/onboarding/context';
+import EmbeddedKyc from 'wcpay/onboarding/steps/embedded-kyc';
+import Logo from 'assets/images/woopayments.svg';
+import { closeSmall, Icon } from '@wordpress/icons';
```
Comment (oaratovskyi) [inline client/onboarding/kyc/index.tsx]: move to external section

**Review by dmallory42 — APPROVED** [review]
Tested using the instructions and it's working well! I even tried leaving the KYC multiple times and coming back and it still remembered where I was.

Great work! Let's merge this into the main Embedded KYC branch.

**Review by vladolaru — APPROVED** [review]
The code changes LGTM and the testing instructions check out!

**Pull request merged**